android/wolfssljni-ndk-gradle/app/src/main/java/com/wolfssl/wolfssljni_ndk_gradle/MainActivity.java

```diff
@@ -20,111 +22,180 @@ public class MainActivity extends AppCompatActivity {
+                String[] cipherSuites = new String[] {"TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA"};
+                sock.setEnabledCipherSuites(cipherSuites);
+                String[] protocols = new String[]{"TLSv1.2"};
+                sock.setEnabledProtocols(protocols);
+
+                ts[3] = System.currentTimeMillis();
+                appendDisplayText("D. Starting Handshake\n");
+                sock.startHandshake();
+
+                ts[4] = System.currentTimeMillis();
+                appendDisplayText("E. Sending GET Request\n");
+                String getCmd = "GET / HTTP/1.1\r\n" +
+                                "Host: www.wolfssl.com\r\n" +
+                                "Accept: */*\r\n\r\n";
+                sock.getOutputStream().write(getCmd.getBytes());
+
+                ts[5] = System.currentTimeMillis();
+                long sizeLimit = 10 * 1024;
+                appendDisplayText("F. Receiving at least " + sizeLimit + " bytes\n");
+                byte[] data = new byte[4 * 1024];
+                ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+                long totalRead = 0;
+                /* Read the first 10kB */
+                while (totalRead < sizeLimit) {
+                    int nRead = sock.getInputStream().read(data, 0, data.length);
+                    if (nRead < 0) break;
+                    buffer.write(data, 0, nRead);
+                    totalRead = totalRead + nRead;
+                }
 
-                    } catch (Exception e) {
-                        tv.append("TLS Connection Failed");
-                        e.printStackTrace();
-                    }
+                ts[6] = System.currentTimeMillis();
+                appendDisplayText("G. Closing socket\n");
+                sock.close();
+
+                ts[7] = System.currentTimeMillis();
+                if ((connectCount % 2) == 0) {
+                    appendDisplayText("H. Disabling hardware offload\n");
+                    wolfProv.setDevId(WolfSSL.INVALID_DEVID);
+                    wolfProv.unRegisterDevId(devId);
+                } else {
+                    appendDisplayText("H. Software crypto still enabled\n");
                 }
-            });
 
-        }
+                ts[8] = System.currentTimeMillis();
+                appendDisplayText("I. Removing Provider\n");
+                Security.removeProvider(wolfProv.getName());
+
+                ts[9] = System.currentTimeMillis();
+
+                appendDisplayText("TLS Connection Successful! Got" +
+                        totalRead + " bytes.\n");
+                appendDisplayText("Times(ms):" +
+                        " A:" + (ts[1] - ts[0]) +
+                        " B:" + (ts[2] - ts[1]) +
+                        " C:" + (ts[3] - ts[2]) +
+                        " D:" + (ts[4] - ts[3]) +
+                        " E:" + (ts[5] - ts[4]) +
+                        ", " + getCmd.length() * 1000 / (ts[5] - ts[4]) + "Bps" +
```

Comment:
We might need to check `ts[5] - ts[4]` for zero before dividing here. When testing on a Galaxy tablet here, I'm seeing a sporadic divide by zero error:

```
java.lang.ArithmeticException: divide by zero
    at com.wolfssl.wolfssljni_ndk_gradle.MainActivity.lambda$new$6$com-wolfssl-wolfssljni_ndk_gradle-MainActivity(MainActivity.java:178)
    at com.wolfssl.wolfssljni_ndk_gradle.MainActivity$$ExternalSyntheticLambda4.run(Unknown Source:2)
```